### PR TITLE
Add second passphrase class - Closes #883

### LINKED ISSUE #883: Add SecondSignatureTransaction class
### Description

Extend `BaseTransaction` implement all the specific features of second signature registration transactions.

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### Description

Adds  second passphrase registration class. 

### Review checklist

* The PR resolves #883 
* Merge after #976
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the commit guidelines
* Documentation has been added/updated

### CODE REVIEW

File: packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+			type: 1,
+			fee: SIGNATURE_FEE.toString(),
+			asset: { signature: { publicKey } },
+		};
+
+		if (!passphrase) {
+			return transaction;
+		}
+
+		const transactionWithSenderInfo = {
```
Comment (shuse2) [inline packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts]: i think we don't need this part, if passphrase is set, senderId and publicKey are set by `createBaseTransaction`

File: packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+		return transaction;
+	}
+
+	protected getAssetBytes(): Buffer {
+		const {
+			signature: { publicKey },
+		} = this.asset;
+
+		return publicKey ? hexToBuffer(publicKey) : Buffer.alloc(0);
```
Comment (shuse2) [inline packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts]: i think publicKey is always not undefined?

File: packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+		}
+	}
+
+	public verifyAgainstOtherTransactions(
+		transactions: ReadonlyArray<TransactionJSON>,
+	): TransactionResponse {
+		const errors = transactions
+			.filter(
+				tx =>
+					tx.senderPublicKey === this.senderPublicKey && tx.type === this.type,
```
Comment (shuse2) [inline packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts]: probably `type` check first would be better?

File: packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+			: [];
+
+		errors.push(...assetErrors);
+
+		if (!this.amount.eq(0)) {
+			errors.push(
+				new TransactionError(
+					'Amount must be zero for second signature registration transaction',
+					this.id,
+					'.asset',
```
Comment (shuse2) [inline packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts]: `.amount`?

File: packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+	}
+
+	public validateSchema(): TransactionResponse {
+		const { status, errors: baseErrors } = super.validateSchema();
+		const errors = [...baseErrors];
+		const valid = validator.validate(
+			secondSignatureAssetFormatSchema,
+			this.asset,
+		);
+		const assetErrors = validator.errors
```
Comment (shuse2) [inline packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts:198]: also, check recipientId is empty?

File: packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+		};
+	}
+
+	public apply({ sender }: RequiredSecondSignatureState): TransactionResponse {
+		const { errors: baseErrors, state } = super.apply({ sender });
+		if (!state) {
+			throw new Error('State is required for applying transaction.');
+		}
+		const errors = [...baseErrors];
+		const { sender: updatedSender } = state as { readonly sender: Account };
```
Comment (shuse2) [inline packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts]: do we need type conversion here?

File: packages/lisk-transactions/src/transactions/3_vote_transaction.ts [hunk 7]
```diff
@@ -299,6 +300,19 @@ export class VoteTransaction extends BaseTransaction {
 			);
 		}
 
+		if (
+			this.recipientPublicKey &&
+			this.recipientId !== getAddressFromPublicKey(this.recipientPublicKey)
+		) {
+			errors.push(
```
Comment (shuse2) [inline packages/lisk-transactions/src/transactions/3_vote_transaction.ts:307]: 👍

**Review by shuse2 — DISMISSED** [review]

**Review by SargeKhan — APPROVED** [review]
Besides the comments left by @shuse2 , the PR looks good!

**Review by shuse2 — APPROVED** [review]

**Pull request merged**